tests/rptest/tests/controller_upgrade_test.py

```diff
@@ -30,12 +25,11 @@ def test_updating_cluster_when_executing_operations(self):
         Validates that cluster is operational when upgrading controller log
         '''
 
-        # set redpanda logical version to value without __consumer_offsets support
-        self.redpanda = RedpandaService(
-            self.test_context,
-            5,
-            # set redpanda logical version to v22.1 - last version without serde encoding
-            environment={"__REDPANDA_LOGICAL_VERSION": 3})
+        # set redpanda version to v22.1 - last version without serde encoding
+        self.redpanda = RedpandaService(self.test_context, 5)
+
+        installer = self.redpanda._installer
+        installer.install(self.redpanda.nodes, (22, 1, 4))
 
         self.redpanda.start()
```

Comment:
nit: is it possible to put any of the above code in ducktape `setUp` phase?
That way setup steps can be separated from the actual upgrade test. 

Or is the service startup here relevant to the test as well? This code reads as though we want the initial version to be 22.1.4.

I recognize my questions are due to a poor understanding of what it means to execute "upgrade tests"

Reply:
You're right, the goal of this snippet is to get all the nodes onto 22.1.4. I could put this in a `setUp()` method, but it wouldn't make much of a difference.

Keep in mind though that `EndToEndTest` (unlike `RedpandaTest`) doesn't override a `setUp()` method and relies on the test body to start Redpanda as I'm doing here, so it doesn't change much whether I put it here or override `setUp()` and do it there.